Src/Newtonsoft.Json/JsonTextReader.cs

```diff
@@ -54,6 +56,8 @@ internal enum ReadType
     /// <summary>
     /// Represents a reader that provides fast, non-cached, forward-only access to JSON text data.
     /// </summary>
+    [RequiresUnreferencedCode(MiscellaneousUtils.TrimWarning)]
+    [RequiresDynamicCode(MiscellaneousUtils.AotWarning)]
```

Comment:
Why add these annotations to types like JsonTextReader? I don't think it uses reflection or other problem APIs.

Same with reader, Newtonsoft.Json.Linq types, etc.

Reply:
I would like to tell people not to use anything to do with the serializer in AOT. But some Newtonsoft.Json things work and don't need warnings on them.

* Reader/writer should be fine in AOT.
* Almost all of JObject, JArray, JValue, etc should almost completely work. Those types have an `ToObject<T>()` method that invokes the serializer and should be annotated individually.

Reply:
Sorry, it's been a while since I've thought about trimming and AOT. I don't know what the feature switches mean. Maybe we should have a call sometime and you can talk through this PR and show me.